Fixes IntelliJ import for 1.4.x

eatkins opened the pull request:
@jastice reported in https://github.com/sbt/sbt/issues/5784 that intellij import is broken with 1.4.0-M2. While debugging this, I found that the jline3 line reader was causing problems for this use case. It was fixable by reverting back to jline 2 when `System.console == null`. My thinking is that we are using jline3 solely for the enhanced tab completions so if `System.console == null` we can assume that we don't need to support enhanced tab completions for that use case. If this assumption is violated, I added an explicit opt in for jline 3 by setting `-Dsbt.jline.version=3`.

eatkins committed:
Honor formatEnabledInEnv at the Terminal level

The sbt.log.noformat parameter should be treated very similarly to
sbt.io.virtual. When it is true, we should just use the raw io streams
for the process. This came up because of
https://github.com/sbt/sbt/issues/5784 which reported that intellij
imports were not working and that ansi control characters were being
written to the output.

eed3si9n commented:
AppVeyor doesn't look happy:

```
[info] [info] Received input event: ShowOptions.
[info] [info] Options:
[info] [info]   <ctrl-d> : interrupt (exits sbt in batch mode)
[info] [info]   <enter>  : interrupt (exits sbt in batch mode)
[info] [info]   '?'      : print options
[info] [info]   'l'      : reload the build
[info] [info]   'q'      : quit sbt
[info] [info]   'r'      : re-run the command
[info] [info]   's'      : return to shell
[info] [info] Received input event: ShowOptions.
[info] [info] Options:
[info] [info]   <ctrl-d> : interrupt (exits sbt in batch mode)
[info] [info]   <enter>  : interrupt (exits sbt in batch mode)
[info] [info]   '?'      : print options
[info] [info]   'l'      : reload the build
[info] [info]   'q'      : quit sbt
[info] [info]   'r'      : re-run the command
[info] [info]   's'      : return to shell
[info] [info] Received input event: ShowOptions.
[info] [info] Options:
[info] [info]   <ctrl-d> : interrupt (exits sbt in batch mode)
....
```

eatkins commented:
Yep! I am going to figure out what is going wrong. Sadly I'll probably need to delve further into the differences between the jline 2 and jline 3 line readers. I'd hoped this could be a simple fix but perhaps it isn't.

eatkins committed:
Fix intellij import with 1.4.x

The intellij import currentlly works by forking an sbt process and
writing command input through the process input stream. To make this
work, we need the SimpleTerminal (which is used when sbt is run with
-Dsbt.log.noformat=true) to be able to read input.

Attaching the input to the simple terminal caused watch tests to fail on
windows. This can be fixed by checking if the byte read from the input
stream is -1 and ignoring it if so.

eatkins commented:
It turns out that I didn't need to revert to a jline 2 line reader to make this work. It was just a matter of correctly wiring up the terminal input stream.
